src/shapes/paint/linear_gradient.cpp

```diff
@@ -82,14 +82,17 @@ void LinearGradient::update(ComponentDirt value) {
         const double ro = opacity() * renderOpacity();
         const auto count = m_Stops.size();
         // TODO: replace these with stack-alloc helpers?
-        ColorInt colors[count];
-        float stops[count];
+        ColorInt* colors = new ColorInt[count];
```

Comment:
Is there a limit on how long "count" can be? If not we probably do want a wrapper that uses stack storage up to a certain threshold, then falls back on heap storage.